src/mavlink_server.cpp

```diff
@@ -303,3 +529,48 @@ void MavlinkServer::stop()
     if (_timeout_handler > 0)
         Mainloop::get_mainloop()->del_timeout(_timeout_handler);
 }
+
+int MavlinkServer::addCameraComponent(CameraComponent *camComp)
+{
+    log_debug("%s", __func__);
+    int id = MAV_COMP_ID_CAMERA;
+    while (id < MAV_COMP_ID_CAMERA6 + 1) {
+        if (compIdToObj.find(id) == compIdToObj.end()) {
+            compIdToObj.insert(std::make_pair(id, camComp));
+            break;
+        }
+        id++;
+    }
+
+    return id;
+}
+
+void MavlinkServer::removeCameraComponent(CameraComponent *camComp)
+{
+    log_debug("%s", __func__);
+
+    if (!camComp)
+        return;
+
+    for (std::map<int, CameraComponent *>::iterator it = compIdToObj.begin();
+         it != compIdToObj.end(); it++) {
+        if ((it->second) == camComp) {
+            compIdToObj.erase(it);
+            break;
+        }
+    }
+
+    return;
+}
+
+CameraComponent *MavlinkServer::getCameraComponent(int compID)
+{
+    if (compID < MAV_COMP_ID_CAMERA || compID > MAV_COMP_ID_CAMERA6)
```

Comment:
Currently 7 component IDs are allotted for camera components in MAVLink. Currently its not an issue as we have less than 7 cameras. Once we hit the number we may need to allot more IDs in the protocol. Technically, there should not be an issue using ID beyond MAV_COMP_ID_CAMERA6 but if QGC put checks for not receiving heartbeats from such out of range components then change in protocol is the only way.